packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,155 @@
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will be marked as sensitive during an active media
+  /// projection session.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_SENSITIVE.
+  sensitive,
+
+  /// The widget tree does not contain sensitive content.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will only be marked as sensitive if other `SensitiveContent`
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_NOT_SENSITIVE.
+  notSensitive,
+
+  /// The sensitivity content level is unknown to Flutter.
+  ///
+  /// This mode may represent the current content sensitivity of the window if, for example, Android
+  /// adds a new mode that is not recognized by the `SensitiveContent` widget.
+  ///
+  /// This mode cannot be used to set the sensitivity level of a `SensitiveContent` widget.
+  _unknown,
+}
+
+/// Service for setting the content sensitivity of the native app window (Android `View`)
+/// that contains the app's widget tree.
+///
+/// This service is only currently supported on Android API 35+.
+class SensitiveContentService {
+  /// Creates service to set content sensitivity of an app window (Android `View`) via
+  /// communication over the sensitive content [MethodChannel].
+  SensitiveContentService() {
+    sensitiveContentChannel = SystemChannels.sensitiveContent;
+  }
+
+  /// The channel used to communicate with the shell side to get and set the
+  /// content sensitivity of an app window (Android `View`).
+  late MethodChannel sensitiveContentChannel;
+
+  /// Sets content sensitivity level of the app window (Android `View`) that contains the app's widget
+  /// tree to the level specified by [contentSensitivity] via a call to the native embedder.
+  Future<void> setContentSensitivity(ContentSensitivity contentSensitivity) async {
+    try {
+      await sensitiveContentChannel.invokeMethod<void>(
+        'SensitiveContent.setContentSensitivity',
+        contentSensitivity.name,
+      );
+    } catch (e) {
+      // Content sensitivity failed to be set.
+      throw FlutterError('Content sensitivity $contentSensitivity failed to be set: $e');
+    }
+  }
+
+  /// Gets content sensitivity level of the app window (Android `View`) that contains
+  /// the app's widget tree.
+  Future<ContentSensitivity> getContentSensitivity() async {
+    String? result;
+    try {
+      result = await sensitiveContentChannel.invokeMethod<String>(
+        'SensitiveContent.getContentSensitivity',
+      );
+    } catch (e) {
+      // Content sensitivity failed to be retrieved.
+      throw FlutterError('Failed to retrieve content sensitivity: $e');
+    }
+
+    final ContentSensitivity contentSensitivity = ContentSensitivity.values.firstWhere(
```

Comment:
If we use int, we don't need to do a linear look up and can just recast with `ContentSensitivity.values[result]`

Reply:
I'm not sold on converting back to ints. See https://github.com/flutter/flutter/pull/158473#discussion_r2006584854.

Reply:
Yeah; we discussed in https://github.com/flutter/flutter/pull/158473#discussion_r2006584854

Reply:
this can just be ContentSensitivity.values[result];